official/resnet/resnet_run_loop.py

```diff
@@ -355,21 +370,35 @@ def resnet_main(flags, model_function, input_function, shape=None):
       allow_soft_placement=True)
 
   # Set up a RunConfig to save checkpoint and set session config.
-  run_config = tf.estimator.RunConfig().replace(save_checkpoints_secs=1e9,
-                                                session_config=session_config)
+  if not flags.use_distribution_strategy:
+    run_config = tf.estimator.RunConfig().replace(
+        save_checkpoints_secs=1e9, session_config=session_config)
```

Comment:
I think the save_checkpoints_secs is a relic that may even be undesireable at this point; the default is 600 secs, which might be more reasonable than...never. In fact, now that I think about it, this explains why the checkpoints collected during training are only ~once per epoch, which seemed too few. (cf https://www.tensorflow.org/api_docs/python/tf/estimator/RunConfig ) Can we remove? Also, I think we can just initialize the RunConfig directly, without .replace. All of that would allow us to simplify this-- distribution=None unless we have these flags set, and we only init the RunConfig once.

Reply:
Fine by me.